xarray/core/dataset.py

```diff
@@ -2643,6 +2643,107 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             result = result._stack_once(dims, new_dim)
         return result
 
+    def to_stacked_array(self, new_dim, dims, variable_dim='variable'):
```

Comment:
Shouldn't we add a ``name`` keyword argument to be consistent with ``to_array``?

Reply:
Sure. added this in 23d9246e.